runtime/runtime/src/lib.rs

```diff
@@ -1097,32 +1114,45 @@ impl Runtime {
         };
         // Refund for the unused portion of the gas at the price at which this gas was purchased.
         let mut gas_balance_refund = safe_gas_to_balance(action_receipt.gas_price, gas_refund)?;
-        let mut gas_deficit_amount = 0;
+
+        let mut gas_deficit = 0;
+        let mut gas_surplus = 0;
```

Comment:
My immediate reaction was: why we maintain two different variables instead of using signed integers (i128 would easily accommodate any gas amount we might want,) but I guess that makes arithmetic below painful…